shared/domains/chelonia/chelonia.js

```diff
@@ -1198,6 +1225,69 @@ export default (sbp('sbp/selectors/register', {
     })
     return msg
   },
+  'chelonia/out/ownResources': async function (contractID: string) {
+    if (!contractID) {
+      throw new TypeError('A contract ID must be provided')
+    }
+
+    const response = await fetch(`${this.config.connectionURL}/ownResources`, {
+      method: 'GET',
+      signal: this.abortController.signal,
+      headers: new Headers([
+        [
+          'authorization',
+          buildShelterAuthorizationHeader.call(this, contractID)
+        ]
+      ])
+    })
+    if (!response.ok) {
+      console.error('Unable to fetch own resources', contractID, response.status)
+      throw new Error(`Unable to fetch own resources for ${contractID}: ${response.status}`)
+    }
+
+    return response.json()
+  },
+  'chelonia/out/deleteContract': async function (
+    contractID: string | string[],
+    credentials: {
+      [contractID: string]: { token: ?string, billableContractID: ?string }
+    } = {}
+  ) {
+    if (!contractID) {
+      throw new TypeError('A contract ID must be provided')
+    }
+    if (!Array.isArray(contractID)) contractID = [contractID]
+    return await Promise.allSettled(contractID.map(async (cid) => {
+      const hasCredential = has(credentials, cid)
+      const hasToken = has(credentials[cid], 'token') && credentials[cid].token
+      const hasBillableContractID = has(credentials[cid], 'billableContractID') && credentials[cid].billableContractID
+      if (!hasCredential || hasToken === hasBillableContractID) {
+        throw new TypeError(`Either a token or a billable contract ID must be provided for ${cid}`)
+      }
+
+      const response = await fetch(`${this.config.connectionURL}/deleteContract/${cid}`, {
+        method: 'POST',
+        signal: this.abortController.signal,
+        headers: new Headers([
+          ['authorization',
+            hasToken
+              // $FlowFixMe[incompatible-type]
+              ? `bearer ${(credentials[cid].token: any).valueOf()}`
+              // $FlowFixMe[incompatible-type]
+              // $FlowFixMe[incompatible-call]
+              : buildShelterAuthorizationHeader.call(this, credentials[cid].billableContractID)]
+        ])
+      })
+      if (!response.ok) {
+        if (response.status === 404 || response.status === 410) {
+          console.warn('Contract appears to have been deleted already', cid, response.status)
+          return
```

Comment:
Why are you logging `Contract appears to have been deleted already` for `404`?

Reply:
The response in this case will be 404 (but also, 404 does mean something maybe was deleted)